pkg/apis/toolchain/v1alpha1/templateupdaterequest_types.go

```diff
@@ -0,0 +1,65 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NSTemplateTierNameLabelKey stores the name of NSTemplateTier that this TemplateUpdateRequest is related to
+const NSTemplateTierNameLabelKey = LabelKeyPrefix + "nstemplatetier"
+
+// TemplateUpdateRequestSpec defines the desired state of TemplateUpdateRequest
+// It contains the new TemplateRefs to use in the MasterUserRecords
+// +k8s:openapi-gen=true
+type TemplateUpdateRequestSpec struct {
+	TierName string `json:"tierName"`
+
+	// The namespace templates
+	// +listType=atomic
+	Namespaces []NSTemplateTierNamespace `json:"namespaces"`
+
+	// the cluster resources template (for cluster-wide quotas, etc.)
+	// +optional
+	ClusterResources *NSTemplateTierClusterResources `json:"clusterResources,omitempty"`
+}
+
+// TemplateUpdateRequestStatus defines the observed state of TemplateUpdateRequest
+// +k8s:openapi-gen=true
+type TemplateUpdateRequestStatus struct {
+	// Conditions is an array of current TemplateUpdateRequest conditions
+	// Supported condition types: ConditionReady
+	// +optional
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=type
+	Conditions []Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// TemplateUpdateRequest is the Schema for the templateupdaterequests API
+// +k8s:openapi-gen=true
+// +kubebuilder:resource:path=templateupdaterequests,scope=Namespaced
+// +kubebuilder:validation:XPreserveUnknownFields
```

Comment:
Maybe worth adding tier name as a print column?
```suggestion
// +kubebuilder:validation:XPreserveUnknownFields
// +kubebuilder:printcolumn:name="Tier",type="string",JSONPath=`.spec.tierName`
```